pkgs/operator/operator.go

```diff
@@ -189,3 +204,27 @@ func rateLimit(logger *zap.Logger, limit int) func(http.Handler) http.Handler {
 		}),
 	)
 }
+
+func processIncomingRequest(logger *zap.Logger, writer http.ResponseWriter, request *http.Request, reqMessageType wire.TransportType, operatorID uint64) (*wire.SignedTransport, error) {
+	rawdata, err := io.ReadAll(request.Body)
+	if err != nil {
+		return nil, fmt.Errorf("operator %d, failed to read request body, err: %v", operatorID, err)
+	}
+	signedMsg := &wire.SignedTransport{}
+	if err := signedMsg.UnmarshalSSZ(rawdata); err != nil {
+		return nil, fmt.Errorf("operator %d, failed to unmarshal SSZ, err: %v", operatorID, err)
+	}
+	// Validate that incoming message has requested type
+	if signedMsg.Message.Type != reqMessageType {
+		return nil, fmt.Errorf("operator %d, received non-init message to init route, err: %v", operatorID, errors.New("not init message to init route"))
```

Comment:
The error seems init specific...
but the from logic seems like it can be generic (not just init message)